.github/workflows/main.yml

```diff
@@ -23,6 +23,26 @@ jobs:
 
       - uses: pre-commit/action@v1.0.1
 
+  build:
```

Comment:
do we want this to happen on PR? I guess this way we know explicitly if a PR breaks our build, which I like.

if so, we'll need it to be a required check after merging